org.eclipse.draw2d/src/org/eclipse/draw2d/FigureCanvas.java

```diff
@@ -300,9 +303,17 @@ private void unhookViewport() {
 
 	private void layoutViewport() {
 		ScrollPaneSolver.Result result;
+		int vBarWidth = 0;
+		int hBarHeight = 0;
+
+		if (!GTK_OVERLAY_SCROLLING_ENABLED) {
```

Comment:
Now that Scrollable.getScrollbarsMode() respects this environment variable, it sounds like it's no longer necessary to check for it in Draw2d.

This has the added benefit that it'll likely also work if overlay scrolling is disabled via e.g. the [gtk-overlay-scrolling](https://docs.gtk.org/gtk3/property.Settings.gtk-overlay-scrolling.html) setting, rather than an environment variable.

Reply:
I've changed the PR. Not sure if we should add a "gtk".equals(SWT.getPlatform()) test since this is the platform the bug was encountered on.

Reply:
> Not sure if we should add a "gtk".equals(SWT.getPlatform()) test since this is the platform the bug was encountered on.

That's probably not necessary. Even if other platforms were to support overlay scrolling, my expectation is that the width and height of the scrollbar should then also be zero.